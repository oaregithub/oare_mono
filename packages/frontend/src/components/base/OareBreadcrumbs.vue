<template>
  <div class="d-flex">
    <div v-for="(item, idx) in items" :key="idx">
      <router-link v-if="item.link" :to="item.link">{{
        item.text
      }}</router-link>
      <span v-else>{{ item.text }}</span>
      <span v-if="idx !== items.length - 1" class="mx-3">{{ '>' }}</span>
    </div>
  </div>
</template>

<script lang="ts">
import { defineComponent, PropType } from '@vue/composition-api';
export interface BreadcrumbItem {
  link: string | null;
  text: string;
}

export default defineComponent({
  name: 'OareBreadcrumbs',
  props: {
    items: {
      type: Array as PropType<BreadcrumbItem[]>,
      required: true,
    },
  },
});
</script>

<style></style>
