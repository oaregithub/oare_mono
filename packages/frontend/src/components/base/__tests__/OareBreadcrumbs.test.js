import Vuetify from "vuetify";
import VueCompositionApi from "@vue/composition-api";
import { mount, createLocalVue } from "@vue/test-utils";
import OareBreadcrumbs from "@/components/base/OareBreadcrumbs.vue";

const vuetify = new Vuetify();
const localVue = createLocalVue();
localVue.use(VueCompositionApi);

describe("OareBreadcrumbs test", () => {
  const breadcrumbItems = [
    {
      link: "/item1",
      text: "Item 1",
    },
    {
      link: "/item2",
      text: "Item 2",
    },
    {
      link: "/item3",
      text: "Item 3",
    },
  ];
  const createWrapper = () =>
    mount(OareBreadcrumbs, {
      localVue,
      vuetify,
      propsData: {
        items: breadcrumbItems,
      },
      stubs: ["router-link"],
    });

  test("matches snapshot", () => {
    expect(createWrapper()).toMatchSnapshot();
  });

  test("shows separator between breadcrumb items", () => {
    const breadcrumbTexts = createWrapper()
      .text()
      .split(">")
      .map((t) => t.trim());

    expect(breadcrumbTexts).toEqual(breadcrumbItems.map((item) => item.text));
  });

  test("does not show > character at end of items", () => {
    const breadcrumbTexts = createWrapper().text().trim();
    expect(breadcrumbTexts.slice(-1)).not.toBe(">");
  });
});
