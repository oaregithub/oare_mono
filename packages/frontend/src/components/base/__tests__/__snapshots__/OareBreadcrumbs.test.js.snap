// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OareBreadcrumbs test matches snapshot 1`] = `
<div class="d-flex">
  <div>
    <router-link-stub to="/item1">Item 1</router-link-stub> <span class="mx-3">&gt;</span>
  </div>
  <div>
    <router-link-stub to="/item2">Item 2</router-link-stub> <span class="mx-3">&gt;</span>
  </div>
  <div>
    <router-link-stub to="/item3">Item 3</router-link-stub>
    <!---->
  </div>
</div>
`;
